### ISSUE TITLE
Configure `sst-env.d.ts` generation

### ISSUE THREAD

**Issue by oscartbeaumont** [sst/sst#4104]
It would be great to be able to disable the generation of `sst-env.d.ts` for a specific package or a group of packages mostly likely by having a configurable exclude pattern.

For example [my monorepo](https://github.com/mattrax/Mattrax) at it's root has:
 - `sst.config.ts`
 - `packages/*`
 - `apps/*`

By the current SST logic everything in `packages/*` gets given an `sst-env.d.ts` even though I would rather it not have one.

In practice I don't want anything in packages directly touching the infrastructure as they are designed to be self-contained packages such as what would potentially be publishable to npm. Not having the types helps to force separatation and cleans up the Git diff everytime a resource with links is changed in SST.

Broken out from my comment in sst/ion#1017.

**Comment by paolostyle** [comment]
I don't want to be annoying but is there any chance this could be addressed one way or another? I'm slowly losing my mind, every other PR has at minimum 30 changed files because we have quite a lot of small packages in the monorepo and because at this point we have a lot of linkable resources, identifying whether I can safely use one in a given package is also becoming very annnoying. As I said, I suspect having each sst-env personalized to a package is going to be difficult and wouldn't solve the issue entirely (e.g. there could be multiple functions, each with a different set of links) but I just really want to gitignore these files and I can't because of CI. `sst generate-types` or something like that is all I ask for.

**Comment by jayair** [comment]
I don't think we can generate types up front. But for your case, would it help if you gitgnored the ones you know didn't matter to you?
